Vates/VatesSimpleGui/ViewWidgets/src/VisibleAxesColor.cpp

```diff
@@ -0,0 +1,130 @@
+
+//#include "MantidKernel/Logger.h"
+
+#include "MantidVatesSimpleGuiViewWidgets/VisibleAxesColor.h"
+
+#include <array>
+#include <vector>
+
+#include "pqActiveObjects.h"
+#include "pqApplicationCore.h"
+#include "pqServerManagerModel.h"
+#include "vtkCommand.h"
+#include "vtkSMDoubleVectorProperty.h"
+#include "vtkSMProperty.h"
+#include "vtkSMPropertyHelper.h"
+#include "vtkSMTransferFunctionProxy.h"
+#include "vtkSMViewProxy.h"
+
+namespace Mantid {
+// static logger
+// Kernel::Logger g_log("VisibleAxesColor");
+
+namespace Vates {
+namespace SimpleGui {
+
+namespace {
+
+void safeSetProperty(vtkSMProxy *gridAxis,
+                     std::initializer_list<const char *> pnames,
+                     const std::array<double, 3> &value) {
+  if (gridAxis) {
+    for (auto pname : pnames) {
+      vtkSMProperty *prop = gridAxis->GetProperty(pname);
+      if (prop) {
+        vtkSMPropertyHelper helper(prop);
+        helper.Set(value.data(), 3);
+        gridAxis->UpdateProperty(pname);
+      }
+    }
+  }
+}
+
+std::array<double, 3> getContrastingColor(const std::vector<double> &color) {
+  double criteria =
+      1. - (0.299 * color[0] + 0.587 * color[1] + 0.114 * color[2]);
```

Comment:
It is a suggested formula for color brightness
https://www.w3.org/TR/AERT#color-contrast